grobid-trainer/src/main/java/org/grobid/trainer/AcknowledgmentTrainer.java

```diff
@@ -0,0 +1,158 @@
+                writer2 = new OutputStreamWriter(os2, "UTF8");
+            }
+
+            // the file for writing the evaluation data
+            OutputStream os3 = null;
+            Writer writer3 = null;
+            if (evalOutputPath != null) {
+                os3 = new FileOutputStream(evalOutputPath);
+                writer3 = new OutputStreamWriter(os3, "UTF8");
+            }
+
+            // get a factory for SAX parser
+            SAXParserFactory spf = SAXParserFactory.newInstance();
+
+            int n = 0;
+            for (; n < refFiles.length; n++) {
+                final File teifile = refFiles[n];
+                String name = teifile.getName();
+                //System.out.println(name);
+
+                final TEIAcknowledgmentSaxParser parser2 = new TEIAcknowledgmentSaxParser();
+
+                // get a new instance of parser
+                final SAXParser p = spf.newSAXParser();
+                p.parse(teifile, parser2);
+
+                final List<String> labeled = parser2.getLabeledResult();
+                totalExamples += parser2.n;
+
+                // add the features
+                String headerAcknowledgment = FeaturesVectorAcknowledgment.addFeaturesAcknowledgment(labeled);
+
+                // format with features for sequence tagging
+                // given the split ratio we write either in the training file or the evaluation file
+                String[] chunks = headerAcknowledgment.split("\n \n");
+
+                for (int i = 0; i < chunks.length; i++) {
+                    String chunk = chunks[i];
+
+                    if ((writer2 == null) && (writer3 != null))
+                        writer3.write(chunk + "\n \n");
+                    else if ((writer2 != null) && (writer3 == null))
+                        writer2.write(chunk + "\n \n");
+                    else {
+                        if (Math.random() <= splitRatio)
+                            writer2.write(chunk + "\n \n");
+                        else
+                            writer3.write(chunk + "\n \n");
+                    }
+                }
+            }
+
+            if (writer2 != null) {
+                writer2.close();
+                os2.close();
+            }
+
+            if (writer3 != null) {
+                writer3.close();
+                os3.close();
+            }
+
+        } catch (Exception e) {
+            throw new GrobidException("An exception occurred while running Grobid.", e);
+        }
+        return totalExamples;
+    }
+
+    /**
+     * Command line execution.
+     *
+     * @param args Command line arguments.
+     * @throws Exception
+     */
+    public static void main(String[] args) throws Exception {
+        GrobidProperties.getInstance();
+        AcknowledgmentTrainer trainer = new AcknowledgmentTrainer();
+        AbstractTrainer.runTraining(trainer);
+        AbstractTrainer.runEvaluation(trainer);
```

Comment:
See comment before. Might not need the evaluation if you use the n-fold validation

Reply:
OK, thanks, I will look at it. 
Do you have any documentation to put me on the right path?
Here:
https://grobid.readthedocs.io/en/latest/Training-the-models-of-Grobid/#n-folds-cross-evaluation, will be enough? or 
here 
https://grobid-quantities.readthedocs.io/en/latest/training.html ?

Reply:
yes, the first link 

in AbstractTrainer there is a method called `evaluateNFold` or something like that (I'm trying to remember)